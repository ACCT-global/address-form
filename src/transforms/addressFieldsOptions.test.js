import {
  getFirstLevel,
  getSecondLevel,
  getThirdLevel,
} from './addressFieldsOptions.js'

describe('Transform Address Fields Options', () => {
  it('should get of the first level', () => {
    const firstLevelcountryData = {
      State01: {
        City01: '1',
        City02: '2',
      },
      State02: {
        City03: '3',
      },
    }

    const result = getFirstLevel(firstLevelcountryData)

    expect(result).toMatchSnapshot()
  })

  it('should get map of first level and an array of second level', () => {
    const secondLevelCountryData = {
      State01: {
        City01: '1',
        City02: '2',
      },
      State02: {
        City03: '3',
      },
    }

    const result = getSecondLevel(secondLevelCountryData)

    expect(result).toMatchSnapshot()
  })

  it('should get map of first and second level, and an array of third level', () => {
    const thirdLevelCountryData = {
      State01: {
        City01: {
          Neighborhood1: '1',
          Neighborhood2: '2',
        },
        City02: {
          Neighborhood3: '3',
        },
      },
      State02: {
        City03: {
          Neighborhood1: '4',
        },
      },
    }

    const result = getThirdLevel(thirdLevelCountryData)

    expect(result).toMatchSnapshot()
  })
})
