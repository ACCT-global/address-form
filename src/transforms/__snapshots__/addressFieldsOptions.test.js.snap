// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transform Address Fields Options should get map of first and second level, and an array of third level 1`] = `
Object {
  "State01": Object {
    "City01": Array [
      "Neighborhood1",
      "Neighborhood2",
    ],
    "City02": Array [
      "Neighborhood3",
    ],
  },
  "State02": Object {
    "City03": Array [
      "Neighborhood1",
    ],
  },
}
`;

exports[`Transform Address Fields Options should get map of first level and an array of second level 1`] = `
Object {
  "State01": Array [
    "City01",
    "City02",
  ],
  "State02": Array [
    "City03",
  ],
}
`;

exports[`Transform Address Fields Options should get of the first level 1`] = `
Array [
  "State01",
  "State02",
]
`;
